### ISSUE TITLE
Memory leaks on XML validation against XSD

### ISSUE THREAD

**Issue by octym** [polotek/libxmljs#221]
Hey guys,

I have a simple node.js app that listens on a port for XML requests and then tests the XML against and XSD and replies with true or false.

Unfortunately I think you guys have a memory leak somewhere, because the memory usage grows in a linear fashion:

![libxml-js-memory-graph](https://f.cloud.github.com/assets/6569840/2061355/78420518-8c4d-11e3-9928-d00ac98dc3d4.jpg)

Here is the app sample code on:
- node: 0.10.25
- libxmljs: 0.8.1
- npm: 1.3.x

```javascript
var fs = require('fs'),
	libxml = require('libxmljs'),
	http = require('http');

fs.readFile('xsd.xml', 'utf8', function (err, data) {
	var xsdDoc = libxml.parseXmlString(data);
	http.createServer(function (request, response) {
		var payload = '';
		request.on('data', function(data) {
			payload += data;
		});
		request.on('end', function() {
			var mu = process.memoryUsage();
			console.log('=== [S]', Math.round(mu.rss/1024/1024*100)/100 + 'MB', '===');
			var requestData = JSON.parse(payload);
			var xml = libxml.parseXmlString(requestData.xml);
			xml.validate(xsdDoc);
			response.end();
		});
	}).listen(5000);
});
```

And for testing I have a simple bash script that runs this command in an infinite loop and sleeps 2 seconds after each cycle:
```
curl -X POST -H "Content-Type: application/json" -d @payload.json localhost:5000
````

If I comment this line:
````javascript
xml.validate(xsdDoc);
````
and run the test again, here's what happens:
![rss-no-validation](https://f.cloud.github.com/assets/6569840/2061409/532c078a-8c50-11e3-9f9e-2a69d34cb7d9.jpg)

I made three heap snapshots and saved a log and a csv file.
They are all accessible here: https://www.dropbox.com/sh/9na0skv53wqh7kf/TMr4hn9NNH?lst

Here's the difference between the first and the last heap snapshot:

Snapshot #1:
![01 heap-snapshot-1-summary](https://f.cloud.github.com/assets/6569840/2061398/c6ad6920-8c4f-11e3-9abe-36d8ee12dd83.jpg)

Snapshot #3:
![03 heap-snapshot-3-comparison-vs-1](https://f.cloud.github.com/assets/6569840/2061399/cea0e864-8c4f-11e3-9e05-130578468f95.jpg)

I hope this helps you guys narrow this down.

**Comment by wsallem** [comment]
Hi,
I have same problem i don't find solution :(

I have  node.js v0.12.2 and i verify that xml/xsd validat.
Unfortunately we have a memory leak  because when i comment this line 
```javascript
xmlDoc1.validate(xsdDoc);
```

**Comment by defunctzombie** [comment]
fixed in master and v0.14.3
